apps/studio/pages/api/platform/storage/[ref]/buckets/index.ts

```diff
@@ -22,7 +22,53 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
 }
 
 const handleGet = async (req: NextApiRequest, res: NextApiResponse) => {
-  const { data, error } = await supabase.storage.listBuckets()
+  const {
+    limit: queryLimit,
+    offset: queryOffset,
+    search: querySearch,
+    sortColumn: querySortColumn,
+    sortOrder: querySortOrder,
+  } = req.query
+
+  const limit = queryLimit
+    ? Array.isArray(queryLimit)
+      ? parseInt(queryLimit[0], 10)
+      : parseInt(queryLimit, 10)
+    : undefined
+  const offset = queryOffset
+    ? Array.isArray(queryOffset)
+      ? parseInt(queryOffset[0], 10)
+      : parseInt(queryOffset, 10)
+    : undefined
+  const search = querySearch
+    ? Array.isArray(querySearch)
+      ? querySearch[0]
+      : querySearch
+    : undefined
+  const sortColumnString = querySortColumn
+    ? Array.isArray(querySortColumn)
+      ? querySortColumn[0]
+      : querySortColumn
+    : undefined
+  const sortColumn = ['id', 'created_at', 'name'].includes(sortColumnString || '')
+    ? (sortColumnString as 'id' | 'created_at' | 'name')
+    : undefined
+  const sortOrderString = querySortOrder
+    ? Array.isArray(querySortOrder)
+      ? querySortOrder[0]
+      : querySortOrder
+    : undefined
+  const sortOrder = ['asc', 'desc'].includes(sortOrderString || '')
+    ? (sortOrderString as 'asc' | 'desc')
+    : undefined
+
+  const { data, error } = await supabase.storage.listBuckets({
+    ...(limit ? { limit } : {}),
+    ...(offset ? { offset } : {}),
+    ...(search ? { search } : {}),
+    ...(sortColumn ? { sortColumn } : {}),
+    ...(sortOrder ? { sortOrder } : {}),
+  })
```

Comment:
nit: Thoughts on adding this as its own hook? Its a bit messy in the page like this